src/main/java/com/amazon/opendistroforelasticsearch/sql/query/AggregationQueryAction.java

```diff
@@ -92,11 +91,15 @@ public SqlElasticSearchRequestBuilder explain() throws SqlParseException {
                             }
                         }
                     }
-                    if(select.getRowCount()>0) {
+                    if (select.getRowCount() > 0) {
                         ((TermsAggregationBuilder) lastAgg).size(select.getRowCount());
                     }
                 }
 
+                if (lastAgg instanceof TermsAggregationBuilder && field instanceof ScriptMethodField && select.getRowCount() > 0) {
```

Comment:
Any chance to add this to the `if` block above by refactoring it? 

And with your changes, for both regular and script method field, we set `size`. What about method field?

Reply:
I was thinking about it, but proper fix quickly becomes much larger starts breaking ton of seemingly unrelated integration tests with different output (e.g. JSON response contains old size), so I wanted to make targeted fix for particular customer issue than full-blown refactoring.

When/if we will decide to fix ignored test, then it's going to be implemented

Reply:
@galkk  I think even without changing the logic, keeping the outer `if` for only checking the object type will look cleaner:

```java
if (lastAgg instanceof TermsAggregationBuilder) {
  if (!(field instanceof MethodField) {
    ...
  } else if (field instanceof ScriptMethodField && select.getRowCount() > 0) {
    ...
  }
}
```

Reply:
Thanks, I've got what you've meant.

I think that it could be even simplier (with ||), added the change